lib/Sema/CSDiagnostics.cpp

```diff
@@ -2049,11 +2049,22 @@ bool RValueTreatedAsLValueFailure::diagnoseAsError() {
 }
 
 bool RValueTreatedAsLValueFailure::diagnoseAsNote() {
-  auto overload = getCalleeOverloadChoiceIfAvailable(getLocator());
+  auto *locator = getLocator();
+
+  auto overload = getCalleeOverloadChoiceIfAvailable(locator);
   if (!(overload && overload->choice.isDecl()))
     return false;
 
   auto *decl = overload->choice.getDecl();
+
+  if (locator->isLastElement<LocatorPathElt::ArgumentAttribute>()) {
+    auto argConv = locator->findLast<LocatorPathElt::ApplyArgToParam>();
+    ASSERT(argConv);
```

Comment:
Given this is just an additional note, do we *really* want to crash here if this is null 😅?

Reply:
Yeah, it would mean that the ArgumentAttribute was recorded in non-argument position, which could point to a large problem.